File: backend/src/main/java/com/myerasmusjourney/backend/dto/ExperienceSimpleDTO.java
package com.myerasmusjourney.backend.dto;


import com.myerasmusjourney.backend.enumeration.Category;

import java.time.LocalDate;
import java.util.Collection;

public record ExperienceSimpleDTO (
        Long id,
        LocalDate date,
        Float rating,
        String title,
        String description,
        Collection<Category> categories
){}

File: frontend/shared/src/models/ExperienceSimpleDTO.ts
import type {Page} from "./Page";

export interface ExperienceSimpleDTO{
    id: number;
    title: string;
    date: string;
    rating: number;
    description: string;
    categories: string[]
    cityName: string;
    country: string;
    authorName: string;
}

export interface ExperiencePageDTO {
    content: ExperienceSimpleDTO[];
    page: Page
}